apps/gdalenhance.cpp

```diff
@@ -653,3 +515,202 @@ static CPLErr EnhancerCallback(void *hCBData, int nXOff, int nYOff, int nXSize,
 
     return CE_None;
 }
+
+/************************************************************************/
+/*                      ReadLUTs()                                      */
+/*                                                                      */
+/*               Read a LUT for each band from a file.                  */
+/************************************************************************/
+
+CPLErr ReadLUTs(const char *pszConfigFile, int nBandCount, int nLUTBins,
+                int ***ppapanLUTs, double **ppadfScaleMin,
+                double **ppadfScaleMax)
+{
+    char **papszLines = CSLLoad(pszConfigFile);
```

Comment:
to avoid memory leak in error code paths:
```suggestion
    const CPLStringList aosLines(CSLLoad(pszConfigFile));
```
and rename accordingly references to the variables. You can use aosLines.size() instead of CSLCount()

Reply:
Oddly I'm testing this error path and ASAN isn't picking up anything. Clearly it leaks, though.